src/js/components/modals/__tests__/LanguagePreferenceFormModal-test.js

```diff
@@ -0,0 +1,77 @@
+import React from "react";
+import { shallow } from "enzyme";
+import { LanguagePreferenceFormModalComponent } from "../LanguagePreferenceFormModal";
+import UserSettingsStore from "../../../stores/UserSettingsStore";
+import { SAVED_STATE_KEY } from "../../../constants/UserSettings";
+import FormModal from "../../FormModal";
+
+const defaultLanguage = "en";
+const newLanguage = "zh";
+const mockFormData = { language: newLanguage };
+
+describe("LanguagePreferenceFormModalComponent", function() {
+  beforeEach(function() {
+    UserSettingsStore.setKey(SAVED_STATE_KEY, { language: defaultLanguage });
+  });
+
+  it("renders with default language", function() {
+    const component = shallow(<LanguagePreferenceFormModalComponent />);
+    expect(component).toMatchSnapshot();
+  });
+
+  describe("when submitting", function() {
```

Comment:
:neckbeard: IMO, we should get out of the habit of testing component internals like state and instance methods. An integration test might be a good alternative to the remaining examples in this suite